### PR TITLE
feat(FX-4529): add updateCollection mutation

### PR DESCRIPTION
**Pull request by nickskalkin**
[Jira ticket](https://artsyproduct.atlassian.net/browse/FX-4529)

Add a `updateCollection` mutation

## Request

```graphql
mutation {
	updateCollection(input: {id: "485437bf-7949-43b5-ab5d-f1d7c9b64ffa", name: "test" }) {
		responseOrError {
			... on UpdateCollectionSuccess {
				collection {
					internalID
					name
				}
			}
			
			... on UpdateCollectionFailure {
				mutationError {
					message
				}
			}
		}
	}
}
```

## Success

```graphql
{
	"data": {
		"updateCollection": {
			"responseOrError": {
				"collection": {
					"internalID": "485437bf-7949-43b5-ab5d-f1d7c9b64ffa",
					"name": "test"
				}
			}
		}
	}
}
```

## Error

```graphql
{
	"data": {
		"updateCollection": {
			"responseOrError": {
				"mutationError": {
					"message": "Name can't be blank."
				}
			}
		}
	}
}
```

### CODE REVIEW

File: src/schema/v2/me/updateCollectionMutation.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  InputProps,
+  any,
+  ResolverContext
+>({
+  name: "updateCollection",
+  description: "Update a collection",
+  inputFields: {
+    id: {
+      description: "The internal ID of the collection",
+      type: GraphQLString,
```
Comment (mzikherman) [inline src/schema/v2/me/updateCollectionMutation.ts:61]: This should be required and non-nullable.

File: src/schema/v2/me/updateCollectionMutation.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+import { ResolverContext } from "types/graphql"
+import { CollectionType } from "./collection"
+
+const SuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "UpdateCollectionSuccess",
+  isTypeOf: (data) => data.id,
+  fields: () => ({
+    collection: {
+      type: CollectionType,
+      resolve: async (response) => {
```
Comment (mzikherman) [inline src/schema/v2/me/updateCollectionMutation.ts:21]: I don't think this needs to be async?

**Review by anandaroop — COMMENTED** [review]
👍🏽

One question, but nothing else to add beyond Matt's comments.

File: src/schema/v2/me/updateCollectionMutation.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  mutateAndGetPayload: async (args, context) => {
+    if (!context.updateCollectionLoader) {
+      throw new Error("You need to be signed in to perform this action")
+    }
+
+    try {
+      const response = await context.updateCollectionLoader(args.id, {
+        name: args.name,
+        user_id: context.userID,
+        saves: true,
```
[the remark this answers is not in the document]
Comment (nickskalkin) [reply src/schema/v2/me/updateCollectionMutation.ts:80]: Right, I think we won't need it, thanks!

File: src/schema/v2/me/updateCollectionMutation.ts (hunk 2 above)
Comment (nickskalkin) [reply src/schema/v2/me/updateCollectionMutation.ts:21]: right, thanks!

**Pull request merged**